### DISCUSSION TITLE
Tile groups

### DISCUSSION THREAD

**Discussion by amoffat** [mapeditor/tiled#4098] [Ideas]
Hi! Loving Tiled.

Are there any plans to add tile groups? I would like some tiles to be logically grouped together into an "object", for the purposes of doing something with that information in my game engine. I don't want to use separate layers for each object in the group, though that would be a workaround. It would be much more convenient if the tiles all stay in the layers that they're in, but could be grouped together to designate separate objects.

The way I'm working around it right now is: I have a separate object layer, and the rectangles in that object layer have a custom integer property `layerIdx`. Any tile from `layerIdx` that the rectangle overlaps with, is considered a separate object by my engine.

![image](https://github.com/user-attachments/assets/cc6944b3-793b-4bdd-9b76-8dca9a730d44)

**Comment by bjorn** [comment]
It sounds like your solution to this issue is perfectly fine, at least when you need such grouping to be done on the map itself, which of course provides a lot of flexibility. I'm not sure a native "tile group" feature in Tiled could really improve on this. Was there any particular improvement in the workflow you were thinking about?

**Reply by amoffat** [reply]
The issue with the layers workaround is that there's no way to distinguish between layers that exist for z-indexing, and layers that exist purely for grouping. I suppose you could add a custom property to a "tile group" layer, but now it is getting burdensome to create a layer-group, especially if you have to create dozens of layers (one for each tree, for example).

The workflow I am envisioning is creating a "trees" layer, which has all my tree tiles, and sits above the ground, but underneath say a "clouds" layer. In the trees layer, I select those groups of 9 tiles, ctrl+g, and now they become a tile group which I can easily move, and which are exported in the tsj/tsx as a list of tile indices for my engine to process as 1 object.
